frontend/src/app/shared/directives/file-input.directive.ts

```diff
@@ -0,0 +1,33 @@
+import type { OnDestroy } from '@angular/core';
+import { Directive, EventEmitter, HostListener, Output } from '@angular/core';
+import { Subject } from 'rxjs';
+
+/**
+ * This directive must only be used on file inputs and provides an output similar to `(input)`,
+ * but consistent across all supported browsers.
+ *
+ * @example
+ * ````html
+ * <input type="file" (appFileInput)="processFileList($event)" />
+ * ````
+ */
+@Directive({
+    selector: 'input[appFileInput]',
+})
+export class FileInputDirective implements OnDestroy {
+    destroy$ = new Subject<void>();
```

Comment:
Dust
Is the `destroy$` member required here? I could not find any observers to it.

Reply:
Very good observation!
I'll look into it this evening.

Reply:
I removed the (unused) destroy logic in dced4d474d0dc8d697386b23f6c801266f56b7f9.

I believe it is safe to assume that `@HostListener` automatically unsubscribes and, while `appFileInput` doesn't `complete()` when the directive is destroyed, angular automatically unsubscribes from Outputs in templates.

So there should hopefully be no memory leak.